R/zchunk_L101.nonghg_en_USA_S_T_Y.R

```diff
@@ -0,0 +1,244 @@
+             FILE = "temp-data-inject/L1321.in_EJ_R_cement_F_Y",
+             FILE = "temp-data-inject/L124.in_EJ_R_heat_F_Yh",
+             FILE = "temp-data-inject/L111.Prod_EJ_R_F_Yh",
+             FILE = "emissions/EPA_SO2",
+             FILE = "emissions/EPA_CO",
+             FILE = "emissions/EPA_NOx",
+             FILE = "emissions/EPA_VOC",
+             FILE = "emissions/EPA_NH3"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L101.so2_tgej_USA_en_Sepa_F_Yh",
+             "L101.co_tgej_USA_en_Sepa_F_Yh",
+             "L101.nox_tgej_USA_en_Sepa_F_Yh",
+             "L101.voc_tgej_USA_en_Sepa_F_Yh",
+             "L101.nh3_tgej_USA_en_Sepa_F_Yh",
+             "L101.in_EJ_R_en_Si_F_Yh"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    IEA_flow_sector <- get_data(all_data, "energy/mappings/IEA_flow_sector")
+    IEA_product_fuel <- get_data(all_data, "energy/mappings/IEA_product_fuel")
+    GCAM_sector_tech <- get_data(all_data, "emissions/mappings/GCAM_sector_tech")
+    EPA_tech <- get_data(all_data, "emissions/mappings/EPA_tech")
+    EPA_SO2 <- get_data(all_data, "emissions/EPA_SO2")
+    EPA_CO <- get_data(all_data, "emissions/EPA_CO")
+    EPA_NOx <- get_data(all_data, "emissions/EPA_NOx")
+    EPA_VOC <- get_data(all_data, "emissions/EPA_VOC")
+    EPA_NH3 <- get_data(all_data, "emissions/EPA_NH3")
+
+    L1231.in_EJ_R_elec_F_tech_Yh <- get_data(all_data, "temp-data-inject/L1231.in_EJ_R_elec_F_tech_Yh")
+    L1322.in_EJ_R_indenergy_F_Yh <- get_data(all_data, "temp-data-inject/L1322.in_EJ_R_indenergy_F_Yh")
+    L144.in_EJ_R_bld_serv_F_Yh <- get_data(all_data, "temp-data-inject/L144.in_EJ_R_bld_serv_F_Yh")
+    L154.in_EJ_R_trn_m_sz_tech_F_Yh <- get_data(all_data, "temp-data-inject/L154.in_EJ_R_trn_m_sz_tech_F_Yh")
+    L1322.Fert_Prod_MtN_R_F_Y <- get_data(all_data, "temp-data-inject/L1322.Fert_Prod_MtN_R_F_Y")
+    L1321.in_EJ_R_cement_F_Y <- get_data(all_data, "temp-data-inject/L1321.in_EJ_R_cement_F_Y")
+    L124.in_EJ_R_heat_F_Yh <- get_data(all_data, "temp-data-inject/L124.in_EJ_R_heat_F_Yh")
+    L111.Prod_EJ_R_F_Yh <- get_data(all_data, "temp-data-inject/L111.Prod_EJ_R_F_Yh")
+
+    # Add technology columns and combine all energy driver data into a single dataframe
+    bind_rows(L1322.in_EJ_R_indenergy_F_Yh, L1322.Fert_Prod_MtN_R_F_Y,
+              L1321.in_EJ_R_cement_F_Y, L124.in_EJ_R_heat_F_Yh, L111.Prod_EJ_R_F_Yh,
+              rename(L144.in_EJ_R_bld_serv_F_Yh, sector = service)) %>%
+      mutate(technology = fuel) ->
+      temp
+
+    L154.in_EJ_R_trn_m_sz_tech_F_Yh %>%
+      mutate(technology = fuel,
+             fuel = paste(mode, size.class, sep = "_"),
+             sector = UCD_sector) %>%
+      select(-mode, -UCD_sector, -size.class, -UCD_technology, -UCD_fuel) %>%
+
+      # Bind all together
+      bind_rows(L1231.in_EJ_R_elec_F_tech_Yh, temp) ->
+      L101.in_EJ_R_en_Si_F_Yh
+
+    L101.in_EJ_R_en_Si_F_Yh %>%
```

Comment:
Maybe, but this is a little tricky. It turns out that there are multiple lines for each combination of year and sector, fuel, region, and technology, which means that once we `gather` it, the dataset can't be re-`spread` (an error is thrown if we try). So we may need to pass these data in wide format, although definitely without the Xyears.